src/test_l3_plat_power.c

```diff
@@ -0,0 +1,346 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:*
+* Copyright 2024 RDK Management
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+ * @addtogroup HPK Hardware Porting Kit
+ * @{
+ * @par The Hardware Porting Kit
+ * HPK is the next evolution of the well-defined Hardware Abstraction Layer
+ * (HAL), but augmented with more comprehensive documentation and test suites
+ * that OEM or SOC vendors can use to self-certify their ports before taking
+ * them to RDKM for validation or to an operator for final integration and
+ * deployment. The Hardware Porting Kit effectively enables an OEM and/or SOC
+ * vendor to self-certify their own Video Accelerator devices, with minimal RDKM
+ * assistance.
+ *
+ */
+
+/**
+ * @addtogroup POWER_MANAGER Power Manager Module
+ * @{
+ */
+
+/**
+ * @addtogroup POWER_MANAGER_HALTEST Power Manager HAL Tests
+ * @{
+ */
+
+/**
+ * @defgroup PLAT_POWER_HALTEST_L2 Power Manager HAL Tests L2 File
+ * @{
+ * @parblock
+ *
+ * ### L2 Tests for Power Manager HAL :
+ *
+ * Level 2 unit test cases for all APIs of Power Manager HAL
```

Comment:
L2 Test description is placed. This to be updated with L3 HAL TEST description.

Reply:
Updated as requested.